airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -302,7 +304,14 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
               try {
                 recordSchemaValidator.validateSchema(messageOptional.get().getRecord());
               } catch (final RecordSchemaValidationException e) {
-                LOGGER.warn(e.getMessage());
+                if (validationErrors.size() < 100) {
+                  final Integer exceptionCount = validationErrors.get(e.getMessage());
+                  if (exceptionCount == null) {
+                    validationErrors.put(e.getMessage(), 1);
+                  } else {
+                    validationErrors.put(e.getMessage(), exceptionCount + 1);
+                  }
+                }
```

Comment:
```suggestion
                final Integer exceptionCount = validationErrors.get(e.getMessage());
                if (exceptionCount == null && validationErrors.size() < 100) {
                  validationErrors.put(e.getMessage(), 1);
                } else {
                  validationErrors.put(e.getMessage(), exceptionCount + 1);
                }
```
These conditionals should be combined in this way so that we are still incrementing the exception count for validation errors that are already in the map even after the map hits the 100 cap

Reply:
ooh i see! oops, will fix that

Reply:
@alovew it actually won't work properly if you combine the size check with the null check, i.e. `if (exceptionCount == null && validationErrors.size() < 100)`, because if the size is >= 100, then it will default to the `else` case and will try to add it to the map with `null + 1` which will fail.

Instead you need to put the size check inside of the null branch